netboot/main.go

```diff
@@ -165,15 +171,25 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	}
 	debug("DHCP: boot file URL is %s", bootfile)
 	// check for supported schemes
-	if !strings.HasPrefix(bootfile, "http://") {
-		return fmt.Errorf("DHCP: can only handle http scheme")
+	matched, err := regexp.Match("^(http|https)://", []byte(bootfile))
```

Comment:
we should probably consider mixed and upper case too

Reply:
I see that the same matching is done below in `getClientForBootfile`, so this should be refactored in a function. What about something like this? https://play.golang.org/p/2WAX_oHVg0m .

```

func getScheme(urlstring string) (string, error) {
	u, err := url.Parse(urlstring)
	if err != nil {
		return "", err
	}
	scheme := strings.ToLower(u.Scheme)
	if scheme != "http" && scheme != "https" {
		return "", errors.New("Invalid or unsupported URL scheme. Currently only http and https are supported")
	}
	return scheme, nil
}
```

Reply:
Good call.